### PR TITLE
fix: pass toolchain env, lint_target_headers

### PR DESCRIPTION
**Pull request by peakschris**
This PR has some minor updates that I've found are needed in for our codebase:

1) env-vars required by the compiler and known to the toolchain are automatically passed to clang-tidy. Avoid use of use_default_shell_env when running commands.
2) Allow the lint_target_headers regex to match header paths with mixed \\ and / separators that seem to be produced by clang-tidy.
3) Set an envvar to prevent msys bash from mangling the path
4) Improve flag handling

---

### CODE REVIEW

**Comment by peakschris** [comment]
FYI @alexeagle @jsharpe

File: lint/clang_tidy.bzl [hunk 1]
```diff
@@ -349,6 +391,7 @@ def lint_clang_tidy_aspect(binary, configs = [], global_config = [], header_filt
             passes these as -isystem. Change this to False to pass these as -I, which allows clang-tidy to regard
             them as regular header files.
         verbose: print debug messages including clang-tidy command lines being invoked.
+        mnemonic_suffix: suffix of mneomnic to be used. A prefix of AspectRulesLint is always used.
```
Comment (alexeagle) [inline lint/clang_tidy.bzl:394]: could you make a separate PR for this change pls? lumping multiple independent things into one PR makes it harder to review and less clear what's related.
In that PR I'll probably ask why clang_tidy is the only aspect with a customizable mnemonic.

Comment (peakschris) [reply lint/clang_tidy.bzl]: That's a good suggestion, have done so

**Comment by peakschris** [comment]
@alexeagle @jsharpe I think this is ready for review now

**Review by alexeagle — COMMENTED** [review]
Hey @jsharpe could you take a quick look at this? I'm out of my depth.

File: lint/clang_tidy.bzl [hunk 2]
```diff
@@ -236,21 +280,18 @@ def clang_tidy_action(ctx, compilation_context, executable, srcs, stdout, exit_c
...
     if exit_code:
         env["CLANG_TIDY__EXIT_CODE_OUTPUT_FILE"] = exit_code.path
         outputs.append(exit_code)
-    if (ctx.attr._verbose):
-        env["CLANG_TIDY__VERBOSE"] = "1"
 
     ctx.actions.run_shell(
         inputs = _gather_inputs(ctx, compilation_context, srcs),
         outputs = outputs,
         tools = [executable._clang_tidy_wrapper, executable._clang_tidy],
```
[the remark this answers is not in the document]
Comment (peakschris) [reply lint/clang_tidy.bzl:294]: why is this needed? Does this pass the compiler, or something else?

**Comment by peakschris** [comment]
Thanks jsharpe, I've applied two suggestions and had a question about the third. Much appreciated!

File: lint/clang_tidy.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsharpe) [reply lint/clang_tidy.bzl:296]: It passes all the stdlib files from the compiler toolchain - its needed for fully hermetic toolchains or the case that you have a sysroot setup in the compiler toolchain.

[the remark this answers is not in the document]
Comment (peakschris) [reply lint/clang_tidy.bzl]: thanks, fixed

**Comment by peakschris** [comment]
@jsharpe please would you check two latest commits, we were finding some issues when running at scale

**Comment by jsharpe** [comment]
> @jsharpe please would you check two latest commits, we were finding some issues when running at scale

Just a quick note to say this is still on my radar to look at but I've not had much time to do so! Will try to in the next week or so.

**Comment by peakschris** [comment]
@jsharpe I know you're busy but we are waiting to consume these fixes, any chance you could have a quick look?

**Comment by jsharpe** [comment]
This seems to be mostly working on linux with RBE / toolchains_llvm - 2 things I run into:
1. Failure to find some C system headers float.h, stddef.h etc.. I suspect this is a toolchain bug though as its picking up the system gcc version of some of these files instead of using the hermetic files from the toolchain_llvm
2. Headers from implementation_deps don't seem to be found.

@peakschris you mentioned you ran into some issues running at scale? What specifically were the issues there?

**Review by jsharpe — APPROVED** [review]
I'm happy with these changes as they are - the issues I've run into with running this on linux can likely be easily fixed in follow-ups so no need to block landing this with those changes.

**Comment by peakschris** [comment]
@jsharpe thanks for the approval

> you mentioned you ran into some issues running at scale? What specifically were the issues there?

With large directories the command line parameter limit was being reached, so one of the changes here was to use param files as much as possible.

**Comment by peakschris** [comment]
@alexeagle please would you approve now that jsharpe has approved?

**Comment by jsharpe** [comment]
@alexeagle another ping on this as I think you were on vacation when the original ping was done so this may have slipped through your notifications.

**Pull request merged**